### PR TITLE
Update checks from latest dist layout changes

### PR DESCRIPTION
**Pull request by devinsba**

### CODE REVIEW

File: src/checks.py [hunk 1]
```diff
@@ -215,7 +216,11 @@ def check_no_binary_files(state: State) -> Optional[str]:
 
 
 def check_build_and_test(state: State) -> Optional[str]:
-    return _check_sh("./mvnw --quiet package", workdir=state.source_dir)
+    return _check_sh(
```
Comment (abesto) [inline src/checks.py:219]: I'd prefer having two separate commands - `_check_sh` can take a list of commands as its first argument, and it'll bail out on the first non-zero exit code (so `_check_sh(["mvn ...", "./mvnw ..."])`). This gives us cleaner output (extra `>> Executing ...` line between the two commands)

**Review by abesto — APPROVED** [review]
One minor comment, otherwise looks good! Feel free to merge after addressing it.

**Comment by abesto** [comment]
Hmm, by the way... wouldn't we also want to verify that `mvnw` is *not* in the source distribution before we download it? Might be simplest to do as a separate test. (Happy to implement it, takes all of two minutes, just looking for confirmation I understand correctly. Also happy to see it added to this PR.)

**Comment by abesto** [comment]
On second thought: the whole way we handle the repo-vs-source-archive verification is not great right now: we have specific things we expect to differ in either direction. I'll give making that more accurate a shot.

**Comment by devinsba** [comment]
Agreed, we need more of a blacklist + ignored list approach

**Comment by abesto** [comment]
@devinsba Check out #3 ;)

File: src/checks.py [hunk 2]
```diff
@@ -215,7 +216,12 @@ def check_no_binary_files(state: State) -> Optional[str]:
...
+        "mvn --quiet -N io.takari:maven:wrapper -Dmaven=3.6.0",
+        "./mvnw --quiet test",
+        "./mvnw --quiet package",
```
Comment (abesto) [inline src/checks.py]: AFAIK `package` includes running tests, so  `test`  is redundant here (we'll run tests twice). Then again, I've been wrong about Maven before. Repeatedly.

Comment (devinsba) [reply src/checks.py]: Yup, I was running them separately to more distinctly know where the failure happens when I did the test locally, I guess it doesn't really matter though, the output does tell you

Comment (abesto) [reply src/checks.py]: Fair enough. I don't have a strong preference here, go forth and merge as you like!

Comment (devinsba) [reply src/checks.py]: I actually, now looking at the lifecycle docs think we might want to run verify, but it looks like that fails locally. I'm not sure we want to fail a release for integration test failures, at least if maybe they are known failures

**Comment by abesto** [comment]
Pers'nally I think we're fine not running integration tests on release verification. Let's trust CI to take care of that, and only do a final sanity check here. Integration tests tend to be finicky and need much environment setup, which I don't think we should try to (a) 100% automate (b) put in the critical path on ASF releases.

Summoning @michaelsembwever to double-check that logic.

File: src/checks.py (hunk 2 above)
Comment (abesto) [reply src/checks.py]: Pers'nally I think we're fine not running integration tests on release verification. Let's trust CI to take care of that, and only do a final sanity check here. Integration tests tend to be finicky and need much environment setup, which I don't think we should try to (a) 100% automate (b) put in the critical path on ASF releases. (Specifically these integration tests are known to fail in some local envs, but work fine on Jenkins)

Summoning @michaelsembwever to double-check that logic.

Comment (abesto) [reply src/checks.py]: See also the conversation starting at https://mail-archives.apache.org/mod_mbox/zipkin-dev/201902.mbox/browser - methinks there was weak consensus that we're OK not running integration tests at release vote verification.

**Comment by devinsba** [comment]
Yeah, this particular integration test suite uses test containers, for some reason it fails from the zip, but is fine when I run from a git checkout, not sure what that's about.

I'm good with this change when you are @abesto

**Comment by abesto** [comment]
Squashed, merged. <3 Thank you!

**Pull request merged**